### PR TITLE
[risk=low][RW-9518] Minor updates to User App API tests and related cleanup

### PR DESCRIPTION
**Pull request by jmthibault79**
Deleted the unused DiskController and DiskConfig
Updated some tests
Removed some unused methods and parameters

Tested locally with no obvious issues

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have added explanatory comments where the logic is not obvious
- [ ] I have run and tested this change locally, and my testing process is described here
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes an API change, I have run the E2E tests on this change against my local server with [yarn test-local](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples) because this PR won't be covered by the CircleCI tests 
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented these in the description
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and updated the appropriate API consumers
  * AoU UI
  * [Perf tests](https://github.com/broadinstitute/mcnulty/blob/develop/src/test/scala/services/AoU.scala)
  * [Researcher Directory export](https://github.com/all-of-us/workbench/wiki/Researcher-Directory-(RDR-export))
  * Cron tasks - for Offline*Controllers
  * SumoLogic - for EgressAlert

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/api/AppsController.java [hunk 1]
```diff
@@ -23,7 +22,6 @@ public class AppsController implements AppsApiDelegate {
   private final LeonardoApiClient leonardoApiClient;
   private final Provider<DbUser> userProvider;
   private final WorkspaceAuthService workspaceAuthService;
-  private final Provider<WorkbenchConfig> workbenchConfigProvider;
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/api/AppsController.java:26]: unused

File: api/src/main/java/org/pmiops/workbench/api/DiskController.java [hunk 2]
```diff
@@ -1,104 +0,0 @@
-package org.pmiops.workbench.api;
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/api/DiskController.java:1]: no longer needed, now that we have migrated to DisksController

File: api/src/main/java/org/pmiops/workbench/api/DisksController.java [hunk 3]
```diff
@@ -38,17 +36,14 @@ public class DisksController implements DisksApiDelegate {
   private final LeonardoApiClient leonardoNotebooksClient;
   private final LeonardoMapper leonardoMapper;
   private final WorkspaceService workspaceService;
-  private final Provider<DbUser> userProvider;
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/api/DisksController.java:41]: unused

File: api/src/main/java/org/pmiops/workbench/utils/mappers/LeonardoMapper.java [hunk 4]
```diff
@@ -114,8 +113,6 @@ default void addPdCloudServiceEnum(
     leonardoGceWithPdConfig.setCloudService(LeonardoGceWithPdConfig.CloudServiceEnum.GCE);
   }
 
-  DiskConfig toDiskConfig(LeonardoDiskConfig leonardoDiskConfig);
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/utils/mappers/LeonardoMapper.java:117]: unused

File: api/src/test/java/org/pmiops/workbench/api/DisksControllerTest.java [hunk 5]
```diff
@@ -104,7 +107,7 @@ public void setUp() {
   }
 
   @Test
-  public void testGetPD() throws ApiException {
+  public void test_getDisk() throws ApiException {
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/api/DisksControllerTest.java:110]: renamed to match the methods they're testing

**Review by yonghaoy — APPROVED** [review]

**Pull request merged**